ltsp/common/info/55-info.sh

```diff
@@ -32,4 +32,6 @@ info_main() {
     list_img_names -v
     printf "\nIMAGES:\n"
     list_img_names -i
+    printf "\nCMDLINE_BOOT_METHODS:\n"
+    list_img_names -m
```

Comment:
If CMDLINE_BOOT_METHODS is just an ltsp.conf variable, it's not significant enough to be shown as a separate section of `ltsp info`. People can just see it by reading ltsp.conf.

Reply:
Hi @alkisg,
the reasoning behind that is as follows:

Booting via CMDLINE_BOOT_METHOD is special, because in most cases (iSCSI, NBD, RBD), you do not create a squashfs but boot from the respective network block device directly. 

Therefore the theses need to be treated differently by certain tools and for that reason I had to include a specialised list_img_names() option in 55-images.sh.

Boot options coming from CMDLINE_BOOT_METHOD might also coexist with classic images and chroots in the ipxe menu. Therefore I would also list "images" (which they actually are not) separately in ltsp info.

Reply:
The design logic was as follows:

- chroots are the directory-based exports that are listed in iPXE
- images/* are the file-based exports that are listed in iPXE
- VMs are files that contain MBR, partitions, file systems etc; these don't show up in iPXE; they can be used as sources for `ltsp image`

What I propose now is:

- each directory-based export will only be listed once in iPXE, and be booted with NFS or whatever other network file system the sysadmin specifies
- each file-based export will only be listed once in iPXE, and be booted with NFS or NBD or iSCSI or whatever other block file system the sysadmin specifies

Sure, it would be possible to list a squashfs image 10 times in iPXE: bootable over NFS or NBD or iSCSI or AoE... but that would make the code too complicated, and the iPXE menu as well.
Experience from LSP5 says "keep it simple".

So I'd prefer to allow only one boot method for chroots, and one boot method for images. If the sysadmin needs more, he'd need to manually maintain his own ltsp.ipxe file.

Maybe we could name the ltsp.conf variables CHROOT_CMDLINE and IMAGE_CMDLINE, if it makes things more clear.

Reply:
@alkisg, thanks for the clarification.

Please see my comment to topics below.

I believe there is a better way to generalise the setup and still allow chroots, images and even both in parallel. One thing missing though to make this work for everything:

BOOT_CMDLINE_[0-9]*
BOOT_DIR_[0-9]*

As one should be able to specify which boot method applies to the contents of which directory.